include/fizzy/fizzy.h

```diff
@@ -133,6 +133,33 @@ typedef struct FizzyExternalGlobal
     FizzyGlobalType type;
 } FizzyExternalGlobal;
 
+/// External kind.
+typedef enum FizzyExternalKind
```

Comment:
Please compare with https://github.com/WebAssembly/wasm-c-api/blob/master/include/wasm.h#L254-L260.

Reply:
What do you mean? They're quite similar except that I use full word for "function" and list them in the order of the spec.

Reply:
They use `uint8_t` storage type. So it could be used in C++ directly. Or wrap it in C++ for better control.

```cpp
typedef uint8_t wasm_externkind_t;
enum wasm_externkind_enum {
  WASM_EXTERN_FUNC,
  WASM_EXTERN_GLOBAL,
  WASM_EXTERN_TABLE,
  WASM_EXTERN_MEMORY,
};

// C++

enum class ExternalKind : wasm_externkind_t {
  func = WASM_EXTERN_FUNC,
  global = WASM_EXTERN_GLOBAL,
  table = WASM_EXTERN_TABLE,
  memory = WASM_EXTERN_MEMORY,
};
```

This may be done other day, but I'm suggesting exploring the design where some declarations are imported from C to C++ instead of only declaring C equivalents of C++ ones.

Reply:
Hmm, this could have been done for `FizzyValueType`, too :thinking:

Reply:
True. We may go old-school in this PR and then consider conversion. This probably requires off-site discussion.

Reply:
> I use full word for "function"

I like this better.

> list them in the order of the spec.

So is the wasm-c-api not in order with the spec?

Reply:
> > list them in the order of the spec.
> 
> So is the wasm-c-api not in order with the spec?

Yeah, there it's func-global-table-memory.